### PR TITLE
Backport Bitcoin#9424, Bitcoin#10123 and Bitcoin#10153

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
decided to break this one off since 9424 is quite big and could conflict with a lot of stuff. See individual commits. Due to the new setup I can't think of an easy way to do a `-debug dash` so yeah...

### CODE REVIEW

**Comment by UdjinM6** [comment]
```
util.h:130:31: error: enumerator value -2147483648 is outside the range of underlying type 'uint32_t {aka unsigned int}'
         SPORK       = (1 << 31),
```
https://travis-ci.org/dashpay/dash/jobs/531324485#L1871

**Comment by UdjinM6** [comment]
Pls see commits in https://github.com/UdjinM6/dash/commits/pr2918 (note: it's based on 31570c2, so you have to drop ab74a79 if you want to cherry-pick them without conflicts).

Travis: https://travis-ci.org/UdjinM6/dash/builds/533565523

**Comment by PastaPastaPasta** [comment]
Applied commits

**Comment by UdjinM6** [comment]
missed https://github.com/UdjinM6/dash/commit/966f9dbe82f142b79d1b468eea2ce78726cef53e

**Comment by PastaPastaPasta** [comment]
rebased on develop

File: src/init.cpp [hunk 1]
```diff
@@ -552,11 +552,10 @@ std::string HelpMessage(HelpMessageMode mode)
         strUsage += HelpMessageOpt("-watchquorums=<n>", strprintf("Watch and validate quorum communication (default: %u)", llmq::DEFAULT_WATCH_QUORUMS));
     }
     std::string debugCategories = "addrman, alert, bench, cmpctblock, coindb, db, http, leveldb, libevent, lock, mempool, mempoolrej, net, proxy, prune, rand, reindex, rpc, selectcoins, tor, zmq, "
-                                  "dash (or specifically: chainlocks, gobject, instantsend, keepass, llmq, llmq-dkg, llmq-sigs, masternode, mnpayments, mnsync, privatesend, spork)"; // Don't translate these and qt below
-    if (mode == HMM_BITCOIN_QT)
-        debugCategories += ", qt";
+                                  "dash (or specifically: chainlocks, gobject, instantsend, keepass, llmq, llmq-dkg, llmq-sigs, mnpayments, mnsync, privatesend, spork)"; // Don't translate these and qt below
```
Comment (UdjinM6) [inline src/init.cpp]: `debugCategories` is not used anymore and should be removed https://github.com/bitcoin/bitcoin/pull/9424/files#diff-c865a8939105e6350a50af02766291b7L444

**Comment by PastaPastaPasta** [comment]
Travis building :) possibly @codablock review

**Comment by UdjinM6** [comment]
Yep, Travis is looking ok-ish - 64 bit linux build failed by timout (as usual) but 32 bit succeeded so I think we should be good.

**Comment by codablock** [comment]
Lets wait for #2932 to be merged and then merge this one. I'd like to make sure that Travis actually becomes constantly green again (last time we ignored this, a whole bunch of flaky test failures creeped in)

**Comment by UdjinM6** [comment]
`develop` is green https://travis-ci.org/dashpay/dash/builds/535988472.

**Pull request merged**